src-docs/src/views/search_bar/search_bar_filters.js

```diff
@@ -62,6 +66,69 @@ const items = times(10, (id) => {
 
 const initialQuery = EuiSearchBar.Query.MATCH_ALL;
 
+const CustomComponent = ({ query, onChange }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isOnlySales, setIsOnlySales] = useState(false);
```

Comment:
let's compute from the query instead of holding a piece of state, so it can respond to other filter buttons or user input

```suggestion
  const tagSalesClause = query.getOrFieldClause('tag', 'sales');
  const isOnlySales = !!(tagSalesClause && tagClause.value.length === 1);
```

Reply:
Great idea 👍